lib/compat/wordpress-6.1/get-global-styles-and-settings.php

```diff
@@ -20,11 +20,15 @@ function wp_add_global_styles_for_blocks() {
 		}
 
 		if ( isset( $metadata['name'] ) ) {
-			$block_name = str_replace( 'core/', '', $metadata['name'] );
 			// These block styles are added on block_render.
 			// This hooks inline CSS to them so that they are loaded conditionally
 			// based on whether or not the block is used on the page.
-			wp_add_inline_style( 'wp-block-' . $block_name, $block_css );
+			if ( 0 === strpos( $metadata['name'], 'core/' ) ) {
+				$block_name = str_replace( 'core/', '', $metadata['name'] );
+				wp_add_inline_style( 'wp-block-' . $block_name, $block_css );
+			} else {
+				wp_add_inline_style( 'global-styles', $block_css );
+			}
```

Comment:
Just a different way to write things. Not a blocker, but I find this easier to read:
* Use `str_starts_with()` instead of `0 === strpos()`
* Add a `stylesheet_handle` var
* Remove the `else` condition

```suggestion
			$stylesheet_handle = 'global-styles';
			if ( str_starts_with( $metadata['name'], 'core/' ) ) {
				$block_name        = str_replace( 'core/', '', $metadata['name'] );
				$stylesheet_handle = 'wp-block-' . $block_name;
			}
			wp_add_inline_style( $stylesheet_handle, $block_css );
```